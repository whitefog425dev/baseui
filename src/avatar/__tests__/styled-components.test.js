/*
Copyright (c) 2018 Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/
// @flow

import React from 'react';
import {shallow} from 'enzyme';
import {StyledAvatar, StyledRoot} from '../index.js';

describe('Avatar styled components', () => {
  const styledComponents = [
    [StyledAvatar, 'StyledAvatar'],
    [StyledRoot, 'StyledRoot'],
  ];

  test.each(styledComponents)('default properties', (Component, name) => {
    const component = shallow(<Component />);
    expect(component.instance().getStyles()).toMatchSnapshot(
      `${name} has expected default styles`,
    );
  });

  test.each(styledComponents)('non-default size', (Component, name) => {
    const component = shallow(<Component $size="scale1200" />);
    expect(component.instance().getStyles()).toMatchSnapshot(
      `${name} has expected styles with non-default size`,
    );
  });

  it('StyledRoot failure state', () => {
    const component = shallow(<StyledRoot $didImageFailToLoad />);
    expect(component.instance().getStyles()).toMatchSnapshot(
      'StyledRoot has expected failure state styles',
    );
  });
});
