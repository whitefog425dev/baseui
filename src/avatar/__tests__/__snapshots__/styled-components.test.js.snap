// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar styled components StyledRoot failure state: StyledRoot has expected failure state styles 1`] = `
Object {
  "backgroundColor": "$theme.colors.mono400",
  "borderRadius": "50%",
  "boxSizing": "border-box",
  "display": "inline-block",
  "height": "$theme.sizing.scale1000",
  "width": "$theme.sizing.scale1000",
}
`;

exports[`Avatar styled components default properties: StyledAvatar has expected default styles 1`] = `
Object {
  "borderRadius": "50%",
  "boxSizing": "border-box",
  "display": "block",
  "height": "$theme.sizing.scale1000",
  "width": "$theme.sizing.scale1000",
}
`;

exports[`Avatar styled components default properties: StyledRoot has expected default styles 1`] = `
Object {
  "backgroundColor": "$theme.colors.mono400",
  "borderRadius": "50%",
  "boxSizing": "border-box",
  "display": "inline-block",
  "height": null,
  "width": null,
}
`;

exports[`Avatar styled components non-default size: StyledAvatar has expected styles with non-default size 1`] = `
Object {
  "borderRadius": "50%",
  "boxSizing": "border-box",
  "display": "block",
  "height": "$theme.sizing.scale1200",
  "width": "$theme.sizing.scale1200",
}
`;

exports[`Avatar styled components non-default size: StyledRoot has expected styles with non-default size 1`] = `
Object {
  "backgroundColor": "$theme.colors.mono400",
  "borderRadius": "50%",
  "boxSizing": "border-box",
  "display": "inline-block",
  "height": null,
  "width": null,
}
`;
